test/test_core.py

```diff
@@ -2757,7 +2783,8 @@ def test_pthread_proxying(self, modularize):
   @node_pthreads
   def test_pthread_proxying_cpp(self):
     self.set_setting('PROXY_TO_PTHREAD')
-    self.set_setting('INITIAL_MEMORY=32mb')
+    if not self.has_changed_setting('INITIAL_MEMORY'):
+      self.set_setting('INITIAL_MEMORY=32mb')
```

Comment:
Maybe worth a function that changes a setting to a value if it hasn't been changed?  I.e., sets a new "default". (doesn't need to be in this PR)